game-core/src/main/java/games/strategy/triplea/ui/HeadedUiContext.java

```diff
@@ -57,10 +57,7 @@
   @Override
   protected void internalSetMapDir(final String dir, final GameData data) {
     resourceLoader = ResourceLoader.getMapResourceLoader(dir);
-    if (mapData != null) {
-      mapData.close();
```

Comment:
Removing this kind of call to `close()` is the main impetus for this update.

Reply:
Hmm I think we still have the initial Problem, but on a higher level.
Previously the code did call `mapData.close()` whenever `internalSetMapDir` is executed in order to close the underlying resourceloader that might still exist.
Now MapData is fine, but in case this method is run twice (or more), we have n - 1 unclosed instances of ResourceLoader lying around, so it looks like this code needs to come back, I'm afraid :(

Reply:
I agree to the issue you raised. The code that seemingly needs to come back it seems is a check for the previous resource loader and to close that. The resource loader is passed around a lot and references are kept to it. If we can avoid those references from being stored, we could safely close `resourceLoader` at the end of the `internalSetMapDir` method, which would long-term be more ideal IMHO.